test/e2e/performanceprofile/functests/utils/nodes/nodes.go

```diff
@@ -425,3 +426,68 @@ func GetByCpuCapacity(nodesList []corev1.Node, cpuQty int) []corev1.Node {
 	}
 	return nodesWithSufficientCpu
 }
+
+// This function fetches the cpus siblings associated with cpu core and returns them
+// Also updates the map by deleting the cpu siblings returned
+func GetCpuSiblings(numaCoreSiblings map[int]map[int][]int, coreKey int) []string {
+	var cpuSiblings []string
+	for key := range numaCoreSiblings {
+		for _, c := range numaCoreSiblings[key][coreKey] {
+			cpuSiblings = append(cpuSiblings, strconv.Itoa(c))
+			delete(numaCoreSiblings[key], c)
+		}
+	}
+	return cpuSiblings
+}
+
+//This function Splits the numa Siblings in to multiple Ranges
+//Example if we get core siblings as below:
+//[1 33 2 34 3 35 4 36 5 37 6 38 7 39 8 40 9 41 10 42],
+//the below function splits and returns ranges of 3 consecutive cpus:
+//1-3,4-6,7-9,33-35,36-38,39-41
+func GetRanges(cpuString string) string {
+	res := strings.Split(cpuString, ",")
+	var cpusInt = []int{}
+	for _, v := range res {
+		ret, _ := strconv.Atoi(v)
+		cpusInt = append(cpusInt, ret)
+	}
+	sort.Ints(cpusInt)
+	j := 0
+	for i := 0; i < len(cpusInt); i++ {
+		j = i + 1
+		if j < len(cpusInt) {
+			if (cpusInt[i] + 1) != cpusInt[j] {
```

Comment:
what are we trying to achieve here?  find the first duplicate value and break?

Reply:
1. We first retrieve the cpu siblings from a specific core or from range of core (which can be either from 1 specific Numa node or all numa nodes)
2. Sort the cpu siblings 
3. Read the siblings in serial order, 
  i. We keep looping as long as the next siblings is one greater than the earlier one, if it's not 
  ii. we take the starting index and index  where we stopped and put in in range form (Cpu(x)- Cpu(y))
  ii. Repeat i, ii  till we are end of the loop.